packages/components/forms/src/BaseInput/BaseInput.styles.ts

```diff
@@ -1,100 +1,140 @@
+      padding: tokens.spacingXs,
+      minHeight: isHighDensity ? tokens.spacingL : tokens.spacingXl,
+      maxHeight: isHighDensity ? tokens.spacingL : tokens.spacingXl,
     };
   }
 
   return {
-    height: '40px',
-    maxHeight: '40px',
+    padding: isHighDensity ? tokens.spacingXs : `10px ${tokens.spacingS}`,
+    minHeight: isHighDensity ? tokens.spacingXl : '40px',
+    maxHeight: isHighDensity ? tokens.spacingXl : '40px',
   };
 };
 
 const getZIndex = ({
   isDisabled,
   isInvalid,
   zIndexBase = tokens.zIndexDefault,
+}: {
+  isDisabled?: boolean;
+  isInvalid?: boolean;
+  zIndexBase?: number;
 }) => (isDisabled || isInvalid ? zIndexBase + 1 : zIndexBase);
 
-const getStyles = ({ as, isDisabled, isInvalid, size, resize }) => ({
-  rootComponentWithIcon: css({
-    position: 'relative',
-    display: 'flex',
-    width: '100%',
-    zIndex: getZIndex({ isDisabled, isInvalid }),
-  }),
-  input: css({
-    outline: 'none',
-    boxShadow: tokens.insetBoxShadowDefault,
-    boxSizing: 'border-box',
-    backgroundColor: isDisabled ? tokens.gray100 : tokens.colorWhite,
-    border: `1px solid ${isInvalid ? tokens.red600 : tokens.gray300}`,
-    borderRadius: tokens.borderRadiusMedium,
-    color: tokens.gray700,
-    fontFamily: tokens.fontStackPrimary,
-    fontSize: tokens.fontSizeM,
-    lineHeight: tokens.lineHeightM,
-    padding: `10px ${tokens.spacingS}`,
-    margin: 0,
-    cursor: isDisabled ? 'not-allowed' : 'auto',
-    width: '100%',
-    zIndex: getZIndex({ isDisabled, isInvalid }),
+type getInputStylesProps = Pick<
+  BaseInputInternalProps,
+  'as' | 'isDisabled' | 'isInvalid' | 'size' | 'resize'
+> & {
+  density?: Density;
+};
 
-    // if the input is a textarea, the resize prop is applied and size should be ignored
-    ...(as === 'textarea' ? { resize } : getSizeStyles({ size })),
+const getInvalidOrDisabledStyles = ({
+  isDisabled,
+  isInvalid,
+}: {
+  isDisabled?: boolean;
+  isInvalid?: boolean;
+}) => {
+  if (isDisabled) {
+    return {
+      borderColor: tokens.gray300,
+      boxShadow: 'none',
+    };
+  }
+  if (isInvalid) {
+    return {
+      borderColor: tokens.red600,
+      boxShadow: tokens.glowNegative,
+    };
+  }
+  return {};
+};
 
-    '&::placeholder': {
-      color: tokens.gray500,
+const getStyles = ({
+  as,
+  isDisabled,
+  isInvalid,
+  size,
+  resize,
+  density = 'low',
+}: getInputStylesProps) => {
+  const densityStyles = {
+    low: {
+      borderRadius: tokens.borderRadiusMedium,
+      lineHeight: tokens.lineHeightM,
+      fontSize: tokens.fontSizeM,
     },
-
-    '&:active, &:active:hover': {
-      borderColor: isInvalid
-        ? tokens.red600
-        : isDisabled
-        ? tokens.gray300
-        : tokens.blue600,
-      boxShadow: isInvalid
-        ? tokens.glowNegative
-        : isDisabled
-        ? 'none'
-        : tokens.glowPrimary,
+    high: {
+      borderRadius: tokens.borderRadiusSmall,
+      lineHeight: tokens.lineHeightS,
```

Comment:
We now have typography [tokens](https://forma-36-git-feat-high-density-typography-scale.colorfuldemo.com/tokens/typography) for high densities:

```suggestion
      lineHeight: tokens.lineHeightMHigh,
      fontSize: tokens.fontSizeMHigh
```